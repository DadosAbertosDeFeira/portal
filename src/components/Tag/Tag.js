import React from 'react';
import PropTypes from 'prop-types';

import styles from './Tag.module.scss';

const Tag = ({ children, className }) => (
  <div className={`${styles.tag} py-xs px-sm inline-block ${className}`}>
    <p className={`${styles.text} text-xs`}>{children}</p>
  </div>
);

Tag.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
};

Tag.defaultProps = {
  className: '',
};

export default Tag;
