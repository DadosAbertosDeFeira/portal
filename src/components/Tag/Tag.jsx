import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

const Tag = ({ children, className }) => (
  <div
    className={classNames('bg-[#f8f9fa] py-xs px-sm inline-block', className)}
  >
    <p className="text-[#656f78] text-xs">{children}</p>
  </div>
);

Tag.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
};

Tag.defaultProps = {
  className: '',
};

export default Tag;
