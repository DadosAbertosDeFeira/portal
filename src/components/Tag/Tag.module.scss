.tag {
  background: #f8f9fa;
  border-radius: 5px;
}

.text {
  color: #656f78;
}
