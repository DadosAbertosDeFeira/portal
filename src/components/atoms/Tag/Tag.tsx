import { Box, type BoxProps } from "atoms/Box";
import { Text } from "atoms/Text";
import classNames from "classnames";
import { forwardRef } from "react";

export type TagProps = BoxProps & {
  children: string;
};

export const Tag = forwardRef<HTMLDivElement, TagProps>(function Tag(
  { className, children, ...props },
  ref
) {
  return (
    <Box
      className={classNames("bg-[#f8f9fa] py-xs px-sm inline-block", className)}
      {...props}
      ref={ref}
    >
      <Text className="text-xs text-[#656f78]">{children}</Text>
    </Box>
  );
});
