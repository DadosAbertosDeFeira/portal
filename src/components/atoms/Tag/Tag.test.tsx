import { render } from "@testing-library/react";

import { Tag, type TagProps } from "./Tag";

describe("Tag", () => {
  const makeSut = ({ children = "Tag", ...props }: Partial<TagProps>) => {
    return render(<Tag {...props}>{children}</Tag>);
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment()).toMatchSnapshot();
  });
});
