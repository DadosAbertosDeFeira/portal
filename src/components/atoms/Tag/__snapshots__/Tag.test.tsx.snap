// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-[#f8f9fa] py-xs px-sm inline-block"
  >
    <p
      class="text-xs text-[#656f78]"
    >
      Tag
    </p>
  </div>
</DocumentFragment>
`;
